Update Slack Docs (#5932)
cypress-io/cypress-documentation: docs/guides/cloud/integrations/slack.mdx

- Reduce the time it takes to catch failed tests

<DocsVideo
  src="https://www.youtube.com/embed/hnDPz7dognY?si=PRRy4f5n6f1jzfVJ"
  title="Cypress app for Slack"
/>

## Getting started

You need just three things to get started:

- Cypress Cloud account
- Ability to configure Cypress Cloud integrations - you must be an admin or the owner of your Cypress Cloud organization.
- Ability to authorize Slack apps - by default, Slack allows any workspace member to authorize apps, however workspace owners may require app approvals before installation. See this [Slack help article](https://slack.com/help/articles/202035138-Add-apps-to-your-Slack-workspace) for more details.

## Install the Cypress app for Slack

1. In the Cypress Cloud, visit the Integrations page for the organization you wish to integrate with Slack.
2. Click the **Enable** button in the Slack section.
3. A popup window will be displayed requesting permission for Cypress to access the
   workspace.
4. After you approve the app permissions, you can choose your Slack workspace and channel to
   associate with the installation.
5. The installation is complete!

> Now Cypress will post run results for all projects in your organization to the specified Slack channel.

Reviewer: Can you do the setup without it turning on all results for the entire org to start? Like can you enable it per project to start? Or is this step required to turn it on for an org entirely.

## Slack configuration

The Slack messages can be tailored to match your teams' needs.

> - The Slack app can send results to one or more public or private Slack channels or directly to a specific user

Reviewer: Is 'Slack app' right here?

Author: Slack app came from Slack's documentation: https://api.slack.com/docs/apps

- You can select your notification preferences for when and where run results should be posted.
- The app can be configured at the organization-level and project-level which supports multiple teams working on separate projects.

### Per-Organization vs per-project configuration

The following describes where to go to configure Slack for an organization and a project. The remaining sections will describe the supported configuration options.

#### Organization settings

> To share Slack configuration for an organization:

Reviewer: Is 'share' the right word here? Did you mean 'update' or 'configure'?

> To share Slack configuration for an organization:

Author: ```suggestion
Configure Slack for the organization:
```

1. Navigate to the Integrations page for the organization you wish to configure.
2. Within the Slack integration, click Configure.
3. Configure the app.

#### Project settings

> To share Slack configuration for a project:

Reviewer: 'share'?

> To share Slack configuration for a project:

Author: ```suggestion
Configurate Slack for a project:
```

1. Navigate to the Project Settings page for the project you wish to configure.
2. Click on the Integrations tab within the Project Settings page.
3. Configure the app with the Slack Integration section.

### Slack channels

You can have Cypress Cloud post run results to specific channels. To add a
channel:

1. Navigate to the Slack configuration page
2. Click **Add Slack Channel**.

> 3. You'll see a popup window that allows you to choose the channel or individual associate

Reviewer: 'individual associate', not sure what you mean here. You mean you can post to private messages?

   with the organization.
4. Search for and selecting the channel messages should post to.

Cypress Cloud will begin to post run results to the channel.

### Direct messages

> You can have Cypress Cloud post run results to specific user. To post to a user:

Reviewer: ```suggestion
You can have Cypress Cloud post run results to a specific direct message To post to a direct message:
```

> You can have Cypress Cloud post run results to specific user. To post to a user:

Author: ```suggestion
You can have Cypress Cloud post run results as a direct message to a single user. To post to a direct message:
```

1. Navigate to the Slack configuration page
2. Click **Add Slack Channel**.
3. You'll see a popup window that allows you to choose the channel or individual associate
   with the organization.

> 4. Search for and selecting the user that messages should be sent to.

Reviewer: ```suggestion
4. Search for and select the user that messages should be sent to.
```

> Cypress Cloud will begin to send run results to the this user.

Reviewer: ```suggestion
Cypress Cloud will begin to send run results to this user.
```

only for failing runs. However, you can select any combination of the followings preferences:

- **Failed runs**: will notify on runs with a failed status.
- **Passed runs**: will notify on runs with a passed status.
- **Flaky tests**: will notify on runs that have an identified flaky test.

1. Navigate to Slack configuration page
2. Select the notification preferences for each channel

#### Filter notifications by tag

When configuring at the project-level, you can filter notifications posted to Slack based on a
[tag](/guides/guides/command-line#cypress-run-tag-lt-tag-gt) the run was
recorded with. This can be helpful to only post notifications on certain runs,
like those to a staging or release environment.

1. Navigate to the Slack configuration page
2. Under **Tags**, select the **Tag** dropdown for the desired integration and
   select which tag to filter.

### Mute a channel

If you want Cypress Cloud to temporarily stop posting Slack messages to a
certain channel, you can **Mute** that channel. This allows you to easily pause
and resume notifications for a specific channel without losing the configuration
you've put in place.

1. Navigate to the Slack configuration page

> 2. Under **Actions**, select your **Mute** for each Slack channel you want

Reviewer: ```suggestion
2. Under **Actions**, select **Mute** for each Slack channel you want
```

   muted.

### Remove a Slack channel

You can have Cypress Cloud stop posting notifications to a channel. You can
remove all Slack channels if you'd prefer to disable global notifications
altogether in favor of per-project notifications.

1. Navigate to the Slack configuration page

> 2. Under **Actions**, select your **Delete** for each Slack channel you want

Reviewer: ```suggestion
2. Under **Actions**, select **Delete** for each Slack channel you want
```

2. Within the Slack Integration, click **Configure**.
3. Click **Uninstall Slack Integration** to uninstall the Cypress integration
   for Slack.

## Security and compliance

Review our [Privacy Policy](https://on.cypress.io/privacy-policy) for information about how the Cypress app for Slack collects and
manages your information.

### Authorizations

When authorized, Cypress can:

> - View information about a user’s identity

Reviewer: Can we be more specific here? This sounds really broad. Like, you can see the user's first name/last name? Or their entire profile?

Author: This is what Slack shares when reviewing the app
![Screenshot 2024-09-25 at 10 12 56 AM](https://github.com/user-attachments/assets/96fe891b-ac54-43bd-a988-d68b9a1eb9c6)

- Post messages to specific channels in Slack & direct message conversations
- Show previews of cloud.cypress.io URLs in messages

> - View URLs from cloud.cypress.io

Reviewer: What? I don't understand this. We are authorized to see what data from the user's slack here? We already have urls from our own cloud account...

> ### App scopes

Reviewer: Can you link to some doc that explains the definitions of scopes?

- User token scopes: `links:read`, `links:write`

> - Bot token scopes: `chat:write`, `incoming-webhook`, `links:read`, `links:write`

Author: ```suggestion
- Bot token scopes: [`chat:write`](https://api.slack.com/scopes/chat:write), [`incoming-webhook`](https://api.slack.com/scopes/incoming-webhook), [`links:read`](https://api.slack.com/scopes/links:read), [`links:write`](https://api.slack.com/scopes/links:write)
```
